### DISCUSSION TITLE
Make sure our code keeps working when there are multiple shards

### DISCUSSION THREAD

**Discussion by abacabadabacaba** [aurora-is-near/rainbow-bridge#478] [General]
Currently, our code makes certain assumptions that are valid when there is only one shard but not when there are multiple shards. Before using the bridge with multi-sharded NEAR blockchain, we should check our logic for generating and verifying proofs regarding NEAR blockchain so that it handles the case of multiple shards correctly.

**Comment by alexauroradev** [comment]
@abacabadabacaba, can you please elaborate on the assumptions?
